### ISSUE TITLE
`--full-matrix` doesn't work with `--sparse`

### ISSUE THREAD

**Issue by apcamargo** [bluenote-1577/skani#32]
I'm experimenting with `skani` to cluster plasmid/virus genomes at scale (see [here](https://github.com/apcamargo/pyleiden)). To get diagonal values I started using `--full-matrix`, but I noticed that those values are not showing up when using `--sparse`. Is this intentional?

Also, it could be useful if there was a parameter that enables the diagonal in the output. I understand why this is not enabled by default (https://github.com/bluenote-1577/skani/pull/31), but there are some cases where it is useful to have it. For example:

```bash
curl -L https://ccb-microbe.cs.uni-saarland.de/plsdb/plasmids/download/plsdb.fna.bz2 \
    | seqkit seq --only-id --upper-case \
    > plsdb.fna

skani triangle -t 16 --sparse -i -m 150 -c 30 -s 70 plsdb.fna > skani_output.tsv

awk 'NR>1 && $3>=95 && ($4>=85 || $5>=85) {
    printf("%s\t%s\t%.4f\n", $6, $7, $3 * ($4 > $5 ? $4 : $5) / 10000)
}' skani_output.tsv > edges.tsv

pyleiden edges.tsv clusters.txt
```

In the example above, sequences that don't have ANI and AF higher than the thresholds for any other sequence in the FASTA will not be in the network at all and will be missing from the input. Of course there are other ways of including them, but having the diagonal values (in this case, using `--sparse`) would make things easier.

**Comment by bluenote-1577** [comment]
Hi Antonio,

Thanks for reaching out. 

It is intentional to not put in diagonal values (e.g. 100% ANI and 100% AF) for --sparse. The reason is kind of for backwards-compatibility with `mash triangle -E`, the analogous command. I recognize how this could make your life harder for creating a graph, though. 

I think I can implement a --diagonal option or something like that, to give a diagonal both when --sparse is enabled and when it isn't. I'll get back to you.

Thanks,
Jim

**Comment by apcamargo** [comment]
Ohh, ok. I assumed that `--full-matrix --sparse` would provide the diagonals because using `--full-matrix` on itself does do that. I didn't know of Mash's behaviour.

I guess that an additional parameter to enable the diagonal would solve this.

**Comment by bluenote-1577** [comment]
i released v0.2.2 and now you can specify --diagonal to include diagonal entries (for even --sparse, CC @fplazaonate)

**Comment by apcamargo** [comment]
Thanks, Jim!
